get-library.js

```diff
@@ -0,0 +1,219 @@
+'use strict';
+
+const https = require('https');
+const tar = require('tar');
+const fs = require('fs');
+const sander = require('sander');
+const os = require('os');
+const path = require('path');
+const opn = require('opn');
+
+
+/*
+ * nRF5x-Command-Line-Tools (nrfjprog) is required for programming. This script
+ * downloads nrfjprog for the current platform.
+ *
+ * On Linux/macOS, the nrfjprog libraries will be placed in pc-nrfjprog-js's directory,
+ * so that nodejs can find them. The script extracts
+ * the nrfjprog tar file, and copies the libraries to the correct directory.
+ *
+ * In win32 platforms, this just downloads the the installer and logs a message
+ * to tell the user they should run it.
+ */
+
+const DOWNLOAD_DIR = path.join(__dirname, 'nrfjprog');
+
+const PLATFORM_CONFIG = {
+    linux: {
+        // See https://www.nordicsemi.com/eng/nordic/Products/nRF51822/nRF5x-Command-Line-Tools-Linux64/51386
+        url: 'https://www.nordicsemi.com/eng/nordic/download_resource/51386/27/17243451/94917',
+        destinationFile: path.join(DOWNLOAD_DIR, 'nrfjprog-linux64.tar'),
+        extractTo: path.join(DOWNLOAD_DIR, 'unpacked'),
+        copyFiles: {
+            source: path.join(DOWNLOAD_DIR, 'unpacked', 'nrfjprog'),
+            destination: path.join(DOWNLOAD_DIR, 'lib'),
+            pattern: /\.so/,
+        },
```

Comment:
How about extracting the tar file directly to `path.join(DOWNLOAD_DIR, 'lib')`? Then we could get rid of the `copyFiles` stuff. The tar library has a filter option that can be applied when extracting, so that we can pull out only the .so/.dylib/.h files. If you also modify this to point to `nrfjprog/lib`, then perhaps we don't even have to copy the .h files around:

https://github.com/NordicSemiconductor/pc-nrfjprog-js/blob/f9fd70c1e4e97369d6c312a9723879babef298fa/CMakeLists.txt#L11